generated-code-samples-for-review/language/language_sentiment_text.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by protoc-gen-go_gapic. DO NOT EDIT.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	language "cloud.google.com/go/language/apiv1"
+	languagepb "google.golang.org/genproto/googleapis/cloud/language/v1"
+)
+
+// [START language_sentiment_text]
+
+// sampleAnalyzeSentiment: Analyzing Sentiment in a String
+//
+// textContent: The text content to analyze
+func sampleAnalyzeSentiment(textContent string) error {
+	ctx := context.Background()
+	c, err := language.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// textContent := "I am so happy and joyful."
+	req := &languagepb.AnalyzeSentimentRequest{
+		Document: &languagepb.Document{
+			Source: &languagepb.Document_Content{
+				Content: textContent,
+			},
+			// Available types: PLAIN_TEXT, HTML
+			Type: languagepb.Document_PLAIN_TEXT,
+			// Optional. If not specified, the language is automatically detected.
+			// For list of supported languages:
+			//   https://cloud.google.com/natural-language/docs/languages
+			Language: "en",
+		},
+		// Available values: NONE, UTF8, UTF16, UTF32
+		EncodingType: languagepb.EncodingType_UTF8,
+	}
+	resp, err := c.AnalyzeSentiment(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	// Get overall sentiment of the input document
+	fmt.Printf("Document sentiment score: %v\n", resp.GetDocumentSentiment().GetScore())
```

Comment:
We usually print to an `io.Writer` to make it easier to test the output and avoid having a lot of noise in test logs. It also matches how people would write to an `http.ResponseWriter` in an HTTP handler.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#print-to-an-iowriter

Reply:
Easy to change. But potential problem is testing the samples ===> @beccasaurus

Reply:
No preference from me, but I've only ever seen `fmt.Printf` used in our Go code.

Some examples from across Google Cloud:
 - [Storage](https://cloud.google.com/storage/docs/getting-bucket-information)
 - [Datastore](https://cloud.google.com/datastore/docs/reference/libraries#using_the_client_library)
 - [Speech](https://cloud.google.com/speech-to-text/docs/sync-recognize#speech-sync-recognize-go)
 - [BigQuery](https://cloud.google.com/bigquery/docs/quickstarts/quickstart-client-libraries#displaying_the_query_result)
 - [DLP](https://cloud.google.com/dlp/docs/inspecting-text#inspecting_a_text_string)